src/main/kotlin/com/cognifide/gradle/aem/pkg/tasks/PackageCompose.kt

```diff
@@ -204,6 +209,7 @@ open class PackageCompose : ZipTask(), AemTask {
 
     fun installBundleProject(projectPath: String, options: BundleInstalledBuilt.() -> Unit = {}) {
         installBundleBuilt("$projectPath:${BundleCompose.NAME}", options)
+        dependsOn("$projectPath:${JavaPlugin.TEST_TASK_NAME}")
```

Comment:
@sabberworm no cycle, but unit tests run

Reply:
Does that mean whenever I compose a package with a bundle in its install folder, the JUnit tests of the bundled package are run? I’m strongly opposed to this, most JUnit tests take a long time. Builds should be as fast as possible. Running tests should only be done when the user explicitly asked for it.

Reply:
it's up to you. historically, I have got requests for that feature, because people don't know how to configure it.

choose between:

`installBundleBuilt(":subproject:bundleCompose")` // no tests run
`installBundleProject(":subproject")` // tests also run

Reply:
hmm but if you really want to do not run unit tests he building package, this line is also not ok for you: https://github.com/Cognifide/gradle-aem-plugin/blob/master/src/main/kotlin/com/cognifide/gradle/aem/pkg/PackagePlugin.kt#L61

but still, you could built with `-x check`. or `-x test` :)

if unit tests are running a very long time, they should be separated in my opinion.
what is more, unit tests should be really fast and as of Gradle is incremental tests also should be run only when there are classpath changes (code changes).

Reply:
Even if I had no problems personally with this, it’s still not how gradle does it normally. Assume a java project with subprojects A and B where B’s java sources depend on A’s. Setting these dependencies in gradle won’t run A’s unit tests when building B (even if B’s artifact bundles A). Analogously, including another project’s artifact in a package shouldn’t either. We should not defy people’s expectations in this way (people who are used to gradle, anyway).
For me, running package checks before upload is borderline but OK because it’s the only check there is whereas for java code there’s at least a compilation step involved anyway. If the code compiles I’d say most people want to be able to see what it does on their AEM machines without having to fix all the unit tests that broke in the process. I also don’t want to have to explain to all my devs that the need to add `-x test` every time the invoke gradle.
For testing stuff locally, low turnaround time is of the utmost importance. For the same reason our webpack compiler won’t do TypeScript typechecking, only transpilation. The full typechecker is run as part of the `check` target, which also runs junit tests and everything else.
I think most set ups have a CI environment that runs all the checks for PRs and branches so errors in those steps are caught early enough.

Reply:
hmm, good points, I will consider them and discuss together with @mierzwid

Reply:
Probably I will add a switches, sth like:

`package.compose.bundleTests=true`  // default false
`package.compose.nestedTests=true` // default false

default false, as you are suggesting, to enforce quick turnaround time, but... correctness enforcing could be achieved by turning these on.

Reply:
I’d be OK with defaulting to `true` as longs as they’re configurable.